icu4c/source/common/rbbi_cache.cpp

```diff
@@ -158,11 +159,12 @@ void RuleBasedBreakIterator::DictionaryCache::populateDictionary(int32_t startPo
 
         // We now have a dictionary character. Get the appropriate language object
         // to deal with it.
-        const LanguageBreakEngine *lbe = fBI->getLanguageBreakEngine(c);
```

Comment:
this is surely wrong, LanguageBreakEngine should be immutable and not changed once created, you need to keep it as a const pointer.

Reply:
Fixed